### PR TITLE
Refactor Pagination child component layout styles (HDS-1542)

### PR DESCRIPTION
**Pull request by KristinLBradley**
### :pushpin: Summary
If merged, this PR will apply group-related layout styles for Pagination child components only if they are nested inside the Pagination wrapper component. It will also fix a bug with the --token-pagination-child-spacing-vertical design token that was causing top margins of child components to not be applied.

- **Website Preview:** https://hds-website-git-hds-1542-refactor-pagination-margins-hashicorp.vercel.app/components/pagination?tab=code
- **Showcase Preview:** https://hds-components-git-hds-1542-refactor-paginatio-0662ac-hashicorp.vercel.app/components/pagination#showcase

### :hammer_and_wrench: Detailed description
* Moved layout styles applied to stand alone components to parent selector so they instead will only apply to nested child components
* Fix bug with --token-pagination-child-spacing-vertical token that was causing spacing between components in the wrapped mobile view not to render. (Top margins of components on 2nd line underneath the 1st line with the navigation controls wasn't working.)

### :link: External links

Jira ticket: [HDS-1542](https://hashicorp.atlassian.net/browse/HDS-1542)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [x] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-1542]: https://hashicorp.atlassian.net/browse/HDS-1542?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/app/styles/components/pagination.scss [hunk 1]
```diff
@@ -31,22 +31,47 @@ $pagination-layout-breakpoint: 1000px;
...
+
+  .hds-pagination-info {
+    grid-area: info;
+    justify-self: flex-start;
+    margin-right: var(--token-pagination-child-spacing-horizontal);
+
+    @media screen and (max-width: $pagination-layout-breakpoint) {
+      margin-top: var(--token-pagination-child-spacing-vertical);
+      margin-left: var(--token-pagination-child-spacing-horizontal);
+    }
```
Comment (KristinLBradley) [inline packages/components/app/styles/components/pagination.scss:66]: Originally I was going to just move the margin styles to apply to nested child components vs. stand-alone but I realized other layout styles related to grid-area etc. also should be moved.

File: packages/components/app/styles/components/pagination.scss [hunk 2]
```diff
@@ -31,22 +31,47 @@ $pagination-layout-breakpoint: 1000px;
     flex-wrap: wrap;
     justify-content: center;
   }
+
+  // Apply group layout styles to child components only when used inside pagination wrapper:
```
Comment (didoo) [inline packages/components/app/styles/components/pagination.scss:35]: Super nitpick: in the subsequent code, the order of declarations is: "info" / "nav" / "size-selector", we could follow the same order here

File: packages/components/app/styles/components/pagination.scss (hunk 1 above)
Comment (didoo) [reply packages/components/app/styles/components/pagination.scss:66]: I think is the right choice. It's even more readable now! :)

**Review by didoo — APPROVED** [review]
👍👍👍👍👍

**Comment by KristinLBradley** [comment]
Question: What should the change log include mention of and what level (Major, Minor, Patch) should it be considered?

Changes included: 
* Changes group layout styles to only apply to nested child components instead of stand alone child components.
* Bug fix for --token-pagination-child-spacing-vertical

**Pull request merged**